client/app/utils/numbers.js

```diff
@@ -1,23 +1,121 @@
+import _ from 'lodash';
+import Immutable from 'immutable';
+import numbro from 'numbro';
+import { t } from './i18n';
+import currencies from '../assets/json/currency_iso.json';
+
 const MINIMUM_DISTANCE = 0.1;
 const PRECISION = 2;
 const DECIMAL_BASE = 10;
 
+// Locales, which are not yet supported by numbro, are not included.
+// So, these languages default to en-US formatting.
+// el-GR, es-CL, ca-ES, hr-HR, id-ID, is-IS, km-KH, ms-MY,
+// ro-RO, sw-KE, vi-VN, bg-BG, mn-MN, zh-HK, ka-GE, sl-SI, ti-ER
+const DEFAULT_LOCALE = 'en-US';
+const KNOWN_LOCALES = [
+  'en-US',
+  'da-DK',
+  'de-DE',
+  'en-AU',
+  'en-GB',
+  'es-ES',
+  'fi-FI',
+  'fr-FR',
+  'fr-CA',
+  'it-IT',
+  'ja-JP',
+  'nb-NO',
+  'nl-NL',
+  'pt-BR',
+  'pt-PT',
+  'ru-RU',
+  'sv-SE',
+  'tr-TR',
+  'zh-CN',
+  'en-NZ',
+  'et-EE',
+  'pl-PL',
+  'hu-HU',
+  'cs-CZ',
+  'th-TH',
+  'zh-TW',
+  'sk-SK',
+];
+
+const initializeNumbro = _.memoize((n, locale) => {
```

Comment:
Ah, true. can't use numbro as it clashes with our eslint rules (shadowing). 
I wanted to make this function a bit more functional style than what actually happens here (i.e. passing numbro as a parameter even though it is global object).